pkg/metrics/metrics.go

```diff
@@ -37,48 +51,75 @@ func NewMetric(name string, labelKeys []string, labelValues []string, value floa
 		return nil, errors.New("expected labelKeys to be of same length as labelValues")
 	}
 
-	m := ""
+	m := strings.Builder{}
+
+	m.WriteString(name)
 
-	m = m + name
+	labelsToString(&m, labelKeys, labelValues)
 
-	m = m + labelsToString(labelKeys, labelValues)
+	m.WriteByte(' ')
 
-	m = m + fmt.Sprintf(" %v", value)
+	writeFloat(&m, value)
 
-	m = m + "\n"
+	m.WriteString("\n")
```

Comment:
DIdn't catch this in my first round of review: This could be written as
```Go
m.WriteByte('\n')
```

Reply:
Right, sorry about that. Follow up: https://github.com/kubernetes/kube-state-metrics/pull/572